File: musicShopProject/Model/Products/Product.cs
namespace musicShopProject.Model.Products;

public class Product
{
    public Guid Id { get; }
    public String Name { get; }
    public String Description { get; }
    public ProductStatus Status { get; }
    public Decimal Price { get; }
    public Decimal Weight { get; }
    public Guid CategoryId { get; }
    public Boolean IsHidden { get; }
    public Int32 Quantity { get; }
    public String[] Images {  get; }

    public Product(
        Guid id, String name, String description, 
        ProductStatus status, Decimal price, Decimal weight,
        Guid categoryId, Boolean isHidden, Int32 quantity, String[] images)
    {
        Id = id;
        Name = name;
        Description = description;
        Status = status;
        Price = price;
        Weight = weight;
        CategoryId = categoryId;
        IsHidden = isHidden;
        Quantity = quantity;
        Images = images;
    }
}

File: musicShopProject/ClientApp/src/domain/products/product.ts
import { PrettyDOMOptions } from "@testing-library/react";
import { ProductStatus } from "./enum/ProductStatus";

export class Product {
    constructor(
        public id: string,
        public name: string,
        public description: string,
        public manufacturer: string,
        public status: ProductStatus,
        public price: number,
        public weight: number,
        public categoryId: string,
        public isHidden: boolean,
        public quantity: number,
        public images: string[]
    ) { }
}

export function mapToProduct(data: any):Product {
    return new Product(data.id, data.name, data.description, data.manufacturer,
        data.status, data.price, data.weight, data.categoryId,
        data.isHidden, data.quantity, data.images
    )
}

export function mapToProducts(data:any[]): Product[] {
    return data.map(mapToProduct);
}